server/src/main/java/com/cloud/template/HypervisorTemplateAdapter.java

```diff
@@ -748,9 +748,6 @@ void checkAndRemoveTemplateDetails(VMTemplateVO template) {
     public TemplateProfile prepareDelete(DeleteTemplateCmd cmd) {
         TemplateProfile profile = super.prepareDelete(cmd);
         VMTemplateVO template = profile.getTemplate();
-        if (template.getTemplateType() == TemplateType.SYSTEM) {
-            throw new InvalidParameterValueException("The DomR template cannot be deleted.");
-        }
```

Comment:
```sugestion
        if (template.getTemplateType() == TemplateType.SYSTEM && ! cmd.isForced()) {
            throw new InvalidParameterValueException("The DomR template cannot be deleted.");
        }
```
is what you mean @GaOrtiga ?

Reply:
Yes, and also update the message in the exception.

Reply:
> ```
>         if (template.getTemplateType() == TemplateType.SYSTEM && ! cmd.isForced()) {
>             throw new InvalidParameterValueException("The DomR template cannot be deleted.");
>         }
> ```
> 
> is what you mean @GaOrtiga ?

@DaanHoogland @weizhouapache given that this operation can only be carried out by Root Admins, could we proceed with this proposed implementation?